browser/brave_shields/ad_block_service_browsertest.cc

```diff
@@ -1604,6 +1621,59 @@ IN_PROC_BROWSER_TEST_F(Default1pBlockingFlagDisabledTest, Custom1pBlocking) {
   EXPECT_EQ(profile()->GetPrefs()->GetUint64(kAdsBlocked), 2ULL);
 }
 
+class AdBlockServiceSignedComponentsTest : public AdBlockServiceTest {
+ public:
+  AdBlockServiceSignedComponentsTest() {
+    feature_list_.InitAndEnableFeature(
+        component_updater::kComponentContentsVerifier);
+  }
+  ~AdBlockServiceSignedComponentsTest() override = default;
+
+ private:
+  base::test::ScopedFeatureList feature_list_;
+};
+
+IN_PROC_BROWSER_TEST_F(AdBlockServiceSignedComponentsTest,
+                       SafeResourcesComponentAccess) {
+  const auto components_path =
+      GetTestDataDir().AppendASCII("adblock-components");
+  {
+    auto* resource_provider = InstallDefaultAdBlockResources(
+        components_path.AppendASCII("resources_ok"));
+    base::RunLoop run_loop;
+    resource_provider->LoadResources(
+        base::BindLambdaForTesting([&run_loop](const std::string& resources) {
+          constexpr const char kExpectedResources[] =
+              "[{\"name\":\"brave-fix.js\",\"aliases\":[],\"kind\":{\"mime\":"
+              "\"application/javascript\"}}]";
+          EXPECT_EQ(kExpectedResources, resources);
+          run_loop.Quit();
+        }));
+    run_loop.Run();
+  }
+
+  {
+    auto* resource_provider = InstallDefaultAdBlockResources(
+        components_path.AppendASCII("resources_corrupted"));
+    base::RunLoop run_loop;
+    resource_provider->LoadResources(
+        base::BindLambdaForTesting([&run_loop](const std::string& resources) {
+#if BUILDFLAG(ENABLE_EXTENSIONS)
```

Comment:
why does it matter if extensions are enabled for adblock? It's not an extension

Reply:
Because it uses the chromium code which placed under this flag.